src/__tests__/components/SignIn/SignIn.test.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import renderWithMockedProvider from "../../testHelper";
+import user from "@testing-library/user-event";
+import { renderHook, screen, waitFor } from "@testing-library/react";
+import SignIn from "../../../pages/sign-in";
+import customAxios from "../../../utils/axios";
+import MockAdapter from "axios-mock-adapter";
+import userSignIn from "../../../components/SignIn/useSignIn";
+import { act } from "react-dom/test-utils";
+
+describe("Sign in Page", () => {
+  const account = { username: "vvv@v.com", password: "123qweASD=" };
+
+  beforeAll(() => {
+    mock.reset();
+  });
+
+  it("should render personal sign in page success", () => {
+    renderWithMockedProvider(<SignIn />);
+    expect(screen.getByText("CourtCanva")).toBeInTheDocument();
+    expect(screen.getByText("Sign in to CourtCanva Franchisee")).toBeInTheDocument();
+  });
+
+  it("should show warning when input is empty", async () => {
+    renderWithMockedProvider(<SignIn />);
+    const submitBtn = screen.getByText("Sign In");
+    user.click(submitBtn);
+    expect(await screen.findByText("Username is required")).toBeVisible();
+    expect(await screen.findByText("Password is required")).toBeVisible();
+  });
+
+  it("should form submit", async () => {
+    renderWithMockedProvider(<SignIn />);
+    const submitBtn = screen.getByText("Sign In");
+    const usernameInput = screen.getByRole("email");
+    const passwordInput = screen.getByRole("password");
+    await user.type(usernameInput, account.username);
+    await user.type(passwordInput, account.password);
+    user.click(submitBtn);
+    await waitFor(() => expect(screen.getByRole("status")).toBeInTheDocument());
```

Comment:
all the combinations of `waitFor` and `getBy...` can be replaced by `findBy`

Reply:
done